dev/releases/HelpLinks-to-JSON.g

```diff
@@ -0,0 +1,53 @@
+##  
+##  A utility for dumping links to all available manual sections 
+##  (for use with named manual references on web pages or in search
+##  utilities).
+##
+##  This is meant to be used with a plain vanilla GAP release, with
+##  only the bundled packages and no special user config.
+##
+
+LoadPackage("json");
+
+InstallMethod(_GapToJsonStreamInternal,
+[IsOutputStream, IsObject],
+function(o, x)
+    PrintTo(o, "null");
+end);
+
+out := OutputTextUser();
+r := rec();
+# load all books (without prompting the user for input)
+Perform(HELP_KNOWN_BOOKS[1], HELP_BOOK_INFO);;
+for x in Set(NamesOfComponents(HELP_BOOKS_INFO)) do # Any point doing Set?
+  book := HELP_BOOKS_INFO.(x);
+
+  # <x> is the lowercase bookname, we want two capitalised versions:
+  # - One with " (not loaded)" removed (we don't want this string in the end)
+  # - One with ": " added (for matching and removing this as a substring)
+  bname := ReplacedString(book.bookname, " (not loaded)", "");
+  match := Concatenation(book.bookname, ": ");
+
+  r.(bname) := rec();
+  for i in [1 .. Length(book.entries)] do
+    entry := HELP_BOOK_HANDLER.HelpDataRef(book, i);
+    name := StripEscapeSequences(entry[1]);
+    name := ReplacedString(name, match, "");
+    NormalizeWhitespace(name);
+
+    path := entry[6];
+    if path = fail then
+      path := "FAIL";
+    else
+      for root in GAPInfo.RootPaths do
+        path := ReplacedString(path, root, "");
```

Comment:
Perhaps also add a comment explaining this loop? I think it ensures that the paths do not containing anything specific to the build machine / are relative. Anyone working on this code in thwnfugure migjt appreciate being told :-)

Reply:
I will add a comment. Yes, the GAP website expects to get paths relative to a GAP root directory, this *should* be able to be achieved by removing any GAP root from the path. I copied this code from you! https://github.com/gap-system/gap/commit/efe19e5284012cfae069c0f38d73d2de95445679#diff-526d642e5b9cfac653edfcfa495f06aa6a1f4f96dea9b26e352fa6f75a221252